app/admin/instances/[id]/page.tsx

```diff
@@ -8,48 +11,117 @@ import DeleteInstanceButton from "@/components/delete-instance-button";
 import CopyableText from "@/components/copyable-text";
+import AgentVersionDisplay from "@/components/agent-version-display";
 import { AWS_DEFAULT_REGION } from "@/server/constants";
 
-export default async function InstanceDetailPage({
+interface HealthStatus {
+  status: "healthy" | "unhealthy" | "loading" | "error" | "offline";
+  error?: string;
+  diagnostics?: {
+    ssm?: {
+      registered?: boolean;
+      pingStatus?: string;
+      lastPingDateTime?: Date;
+      timeSinceLastPing?: string;
+      platformType?: string;
+      agentVersion?: string;
+      associationStatus?: string;
+      lastAssociationExecutionDate?: Date;
+    };
+    healthcheck?: {
+      environmentVariables?: {
+        critical?: Record<string, boolean>;
+        optional?: Record<string, boolean>;
+      };
+      processes?: string;
+      memory?: string;
+      fatalErrors?: boolean;
+    };
+    networkError?: {
+      message: string;
+      type: string;
+      statusCode?: number;
+      statusText?: string;
+      timing?: number;
+      connectivity: {
+        healthCheckApi: boolean;
+        basicConnectivity: boolean;
+      };
+      attemptedUrl?: string;
+      errorName?: string;
+      errorCode?: string | number;
+      timedOut?: boolean;
+    };
+    memoryDiagnostics?: {
+      hasMemoryIssues: boolean;
+      memoryErrors: string[];
+      consoleOutputError?: string;
+    };
+  };
+}
+
+export default function InstanceDetailPage({
   params,
 }: {
   params: Promise<{ id: string }>;
 }) {
-  const { id } = await params;
-  const ec2 = new EC2({
-    region: AWS_DEFAULT_REGION,
-  });
-
-  let instance: {
+  const [id, setId] = useState<string>("");
+  const [healthData, setHealthData] = useState<HealthStatus | null>(null);
+  const [instance, setInstance] = useState<{
     InstanceId?: string;
     State?: { Name?: string };
     KeyName?: string;
     PublicDnsName?: string;
     InstanceType?: string;
     ImageId?: string;
     Tags?: Array<{ Key?: string; Value?: string }>;
-  } | null = null;
-  let errorMessage: string | null = null;
-
-  try {
-    const result = await ec2.describeInstances({
-      InstanceIds: [id],
-    });
-
-    const instances =
-      result.Reservations?.flatMap((r) => r.Instances || []) || [];
-    instance = instances[0] || null;
-
-    if (!instance || !instance.InstanceId) {
-      errorMessage = instance
-        ? `Instance data incomplete - missing required fields`
-        : `Instance with ID "${id}" not found.`;
-    }
-  } catch (error) {
-    console.error("Failed to fetch instance:", error);
-    errorMessage =
-      error instanceof Error ? error.message : "Unknown error occurred";
-  }
+  } | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  useEffect(() => {
+    params.then((p) => setId(p.id));
+  }, [params]);
+
+  useEffect(() => {
+    if (!id) return;
+
+    const fetchInstance = async () => {
+      const ec2 = new EC2({
```

Comment:
I'm seeing this error now on the vercel branch, fix it![image](https://github.com/user-attachments/assets/e5aca7b1-10fd-4069-8c32-8b4d69636955)